internal/test/framework/framework.go

```diff
@@ -294,6 +301,73 @@ func (f *TestFramework) GetNode(externalIP string) (*v1.Node, error) {
 	return matchedNode, nil
 }
 
+// GetNode uses external IP and finds out the name associated with the node
+func (f *TestFramework) GetNodeName(externalIP string) string {
```

Comment:
Why not just return an error here, and then handle the error how you wish in the calling function? You end up checking if  `len(nodeName) == 0` anyway.

I'd also remove the log messages within the function itself and log when you error handle. Doing it this way limits the way this function can be used